NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -308,9 +306,16 @@ def begin(self,
             
                 if not all_files_in_directory(dir_file):
                     self.logger.error(f"Incomplete directory: {dir_file}. Skip ...")
-                    continue      
-            
-                dataset = mattak.Dataset.Dataset(station=0, run=0, data_dir=dir_file, verbose=verbose, **mattak_kwargs)
+                    continue
+                
+                try:
+                    dataset = mattak.Dataset.Dataset(station=0, run=0, data_dir=dir_file, verbose=verbose, **mattak_kwargs)
+                except (ReferenceError, KeyError) as e:
+                    self.logger.error(f"The following exeption was raised reading in the run: {dir_file}. Skip that run ...:\n")
+                    import traceback
+                    traceback.print_exc()
+                    print("")
+                    continue
```

Comment:
I'm happy. But out of curiosity, is there any reason to prefer this structure (importing traceback and printing the exception) over passing the exception to the logger, using `exc_info=e`? See https://docs.python.org/3/library/logging.html#logging.Logger.debug.

Reply:
Okay, I've been stubborn enough to change it. I'm not sure if there is a difference (e.g. in the traceback depth) but I think it's tidier to do everything through the logger output as opposed to printing to stdout.

Reply:
Looks good. Thanks for doing that!